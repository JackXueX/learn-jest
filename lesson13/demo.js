// import $ from 'jquery';
const jsdom = require('jsdom');
const $ = require('jquery')(new jsdom.JSDOM().window);

const addDivToBody = () => {
  $('body').append('<div/>');
}

export default addDivToBody;