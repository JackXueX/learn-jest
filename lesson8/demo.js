// export const runCallback = (callback) => {
//   callback();
// }

export const runCallback = (callback) => {
  callback('abc');
}