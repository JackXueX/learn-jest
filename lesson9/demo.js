export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8000,
    domain: 'localhost1',
    time: new Date()
  }
}

export const generateAnotherConfig = () => {
  return {
    server: 'http://localhost2',
    port: 8000,
    domain: 'localhost',
    time: new Date()
  }
}