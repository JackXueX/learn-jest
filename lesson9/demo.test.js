import { generateConfig, generateAnotherConfig } from "./demo";

// test('测试 generateConfig 函数', () => {
//   // expect(generateConfig()).toEqual({
//   //   server: 'http://localhost',
//   //   port: 8000,
//   //   domain: 'localhost'
//   // });

//   expect(generateConfig()).toMatchSnapshot({
//     time: expect.any(Date)
//   });
// });

// test('测试 generateAnotherConfig 函数', () => {
//   expect(generateAnotherConfig()).toMatchSnapshot({
//     time: expect.any(Date)
//   });
// });

test("测试 generateAnotherConfig 函数", () => {
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "port": 8000,
      "server": "http://localhost2",
      "time": Any<Date>,
    }
  `
  );
});

test("测试 generateAnotherConfig 函数", () => {
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "port": 8000,
      "server": "http://localhost2",
      "time": Any<Date>,
    }
  `
  );
});
