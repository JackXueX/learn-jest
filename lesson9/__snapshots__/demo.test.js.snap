// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateAnotherConfig 函数 2`] = `
Object {
  "domain": "localhost",
  "port": 8000,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
